### ISSUE TITLE
[devcontainer over SSH] configured dotfiles are applied, but not reflected in settings UI

### ISSUE THREAD

**Issue by akosyakov** [microsoft/vscode-remote-release#9926]
While over SSH, configured dotfiles are applied to the devcontainer, but not reflected in settings UI.

<img width="272" alt="Screenshot 2024-05-27 at 11 20 51" src="https://github.com/microsoft/vscode-remote-release/assets/3082655/3782f0ff-0436-4cb0-af63-d4605ef9ad26">

Steps to Reproduce:

1. Configure dotfiles in user settings
2. Open devcontainer over remote SSH
3. Dotfiles are applied, but from settings it looks like they are not configured

<img width="1511" alt="local user settings" src="https://github.com/microsoft/vscode-remote-release/assets/3082655/5146f9ae-3be5-4c64-874d-437ca28cc8bf">
<img width="1136" alt="remote user settings" src="https://github.com/microsoft/vscode-remote-release/assets/3082655/aef6b776-67b9-44b4-bac9-2b9bb9be6b20">
<img width="1136" alt="remote machine settings" src="https://github.com/microsoft/vscode-remote-release/assets/3082655/ca277a6d-092b-41da-af87-9b79d4b29ba7">

Does this issue occur when you try this locally?: Yes/No
Does this issue occur when you try this locally and all extensions are disabled?: Yes/No

**Comment by chrmarti** [comment]
I agree this is confusing. We pick up the Remote setting if there is one, but then fall back to the local setting.

Ideally the UI would allow the user to edit the setting in both the Remote and the User tab of the settings editor, but there is no setting scope supporting this.

The reason this works in the extension has to do with the extension running as a remote resolver when the setting is read.
